interop: op-supervisor panics when L2 CL is stopped and restarted

At commit [9f480717e640591768939ef4cd97bc010f4e2ff0](https://github.com/ethereum-optimism/optimism/commit/9f480717e640591768939ef4cd97bc010f4e2ff0)(most recent commit), with geth: us-docker.pkg.dev/oplabs-tools-artifacts/images/op-geth:4bc345b22fbee14d3162becd197373a9565b7c6d(almost most recent commit) op supervisor panics in the below scenario:

1. Spawn the kt interop devnet: op supervisor and op-node are in sync
2. Stop the op-node container (choose any L2 chain)
3. Restart the op-node container
4. Supervisor panics and halts

Panic log from supervisor
```
crossUnsafe="BlockSeal(hash:0xa31f1109cb960d23358f2ffcddc74e811da35a9a5a19840e6fdcd701d0b339a9, number:100, time:1743179200)"
panic: send on closed channel
	panic: close of closed channel

goroutine 175 [running]:
github.com/ethereum-optimism/optimism/op-service/rpc.SubscribeStream[...].func1()
	/app/op-service/rpc/stream.go:122 +0x130
github.com/ethereum/go-ethereum/event.NewSubscription.func1()
	/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.3.0.20250327000707-c3a989eb882d/event/subscription.go:53 +0x6c
created by github.com/ethereum/go-ethereum/event.NewSubscription in goroutine 179
	/go/pkg/mod/github.com/ethereum-optimism/op-geth@v1.101503.2-rc.3.0.20250327000707-c3a989eb882d/event/subscription.go:51 +0xd8
```

+ when i manually start supervisor again, everything seems to become normal again

--- a/op-supervisor/supervisor/backend/syncnode/node.go
+++ b/op-supervisor/supervisor/backend/syncnode/node.go
@@ -156,6 +156,8 @@ func (m *ManagedNode) SubscribeToNodeEvents() {
 			if prevErr != nil {
 				// This is the RPC runtime error, not the setup error we have logging for below.
 				m.log.Error("RPC subscription failed, restarting now", "err", prevErr)
+				// When the subscription fails, the channel may have been immediately closed
+				m.nodeEvents = make(chan *types.ManagedEvent, 10)
 			}
 			sub, err := m.Node.SubscribeEvents(ctx, m.nodeEvents)
 			if err != nil {

